core/domain/classifier_domain.py

```diff
@@ -169,3 +170,162 @@ def validate(self):
             classifier_registry.Registry.get_classifier_by_algorithm_id(
                 self.algorithm_id))
         classifier_class.validate(self.cached_classifier_data)
+
+
+class TrainClassifierJob(object):
+    """Domain object for a classifier training job.
+
+    A classifier training job is an abstraction of a request made by Oppia
+    for training a classifier model using certain dataset and a particular ML
+    algorithm denoted by the algorithm id. The classifier training jobs are
+    then picked up by the Virtual Machine (VM) through APIs exposed by Oppia.
+    The training_data is populated lazily when the job is fetched from the
+    database upon the request from the VM.
+
+    Attributes:
+        job_id: str. The unique id of the classifier training job.
+        algorithm_id: str. The id of the algorithm that will be used for
+            generating the classifier.
+        exp_id: str. The id of the exploration that contains the state
+            for which the classifier will be generated.
+        exp_version_when_created: str. The version of the exploration when
+            the training job was generated.
+        state_name: str. The name of the state for which the classifier will be
+            generated.
+        status: str. The status of the training job request. This can be either
+            PENDING (default value) or COMPLETE.
+        training_data: list. The training data that is used for training the
+            classifier. This field populated lazily when the job request is
+            picked up by the VM.
+    """
+
+    def __init__(self, job_id, algorithm_id, exp_id, exp_version_when_created,
+                 state_name, status, training_data):
+        """Constructs an TrainClassiferJob domain object.
+
+        Args:
+        job_id: str. The unique id of the classifier training job.
+        algorithm_id: str. The id of the algorithm that will be used for
+            generating the classifier.
+        exp_id: str. The id of the exploration id that contains the state
+            for which classifier will be generated.
+        exp_version_when_created: str. The version of the exploration when
+            the training job was generated.
+        state_name: str. The name of the state for which the classifier will be
+            generated.
+        status: str. The status of the training job request. This can be either
+            PENDING (default value) or COMPLETE.
```

Comment:
@seanlip um yes, thanks! A job can FAIL but it will always be picked up again so it can't actually fail :)

NEW sounds good, it can be used to differentiate between jobs that have been picked up (PENDING) from those that have not been picked up (NEW).